api.lua

```diff
@@ -104,3 +111,45 @@ function mail.send(m)
 
 	return true
 end
+
+function mail.save_draft(m)
+	if type(m.from) ~= "string" then return false, "'from' is not a string" end
+	if type(m.to) ~= "string" then return false, "'to' is not a string" end
+	if type(m.body) ~= "string" then return false, "'body' is not a string" end
+
+	-- defaults
+	m.subject = m.subject or "(No subject)"
+
+	-- limit subject line
+	if string.len(m.subject) > 30 then
+		m.subject = string.sub(m.subject,1,27) .. "..."
+	end
+
+	minetest.log("action", f("[mail] %q saves draft with subject %q and body %q",
+		m.from, m.subject, m.body
+	))
```

Comment:
Also not that significant but I'd argue that this should probably be `verbose` level logging instead of `action`, especially with full mail body.